app/models/miq_vim_broker_worker/runner.rb

```diff
@@ -149,44 +149,58 @@ def drain_event
     event = @queue.deq
     return if event.nil?
 
+    ems_id = ems_ids_for_notify(event[:server], event[:username])
+    return if ems_id.nil?
+
     method = "on_#{event[:op].to_s.underscore}_event"
-    send(method, event) if self.respond_to?(method)
+    send(method, ems_id, event) if respond_to?(method)
   end
 
-  def on_notify_event(event)
-    ems_id = ems_ids_for_notify(event[:server], event[:username])
-    return if ems_id.nil?
-    _log.info("Queueing update for EMS id: [#{ems_id}] on event [#{event[:objType]}-#{event[:op]}]#{" for properties: #{event[:changedProps].inspect}" if event.key?(:changedProps)}")
-    EmsRefresh.queue_vc_update(ems_id, event)
+  OBJ_TYPE_TO_TYPE_AND_CLASS = {
+    'VirtualMachine' => [:vm,   VmOrTemplate],
+    'HostSystem'     => [:host, Host]
+  }.freeze
+
+  def on_create_event(ems_id, event)
+    # TODO: Implement
+    _log.debug("Ignoring refresh for EMS id: [#{ems_id}] on event [#{event[:objType]}-create]")
+    nil
+  end
+
+  def on_delete_event(ems_id, event)
+    # TODO: Implement
+    _log.debug("Ignoring refresh for EMS id: [#{ems_id}] on event [#{event[:objType]}-delete]")
+    nil
   end
-  alias_method :on_create_event, :on_notify_event
-  alias_method :on_delete_event, :on_notify_event
 
-  def on_update_event(event)
-    obj_type, changed_props, change_set = event.values_at(:objType, :changedProps, :changeSet)
+  def on_update_event(ems_id, event)
+    obj_type, mor, changed_props = event.values_at(:objType, :mor, :changedProps)
+
+    type, klass = OBJ_TYPE_TO_TYPE_AND_CLASS[obj_type]
+    return if type.nil?
 
-    type, = EmsRefresh::VcUpdates::OBJ_TYPE_TO_TYPE_AND_CLASS[obj_type]
+    obj = klass.find_by(:ems_ref => mor, :ems_id => ems_id)
+    return if obj.nil?
 
     changed_props.reject! { |p| !ManageIQ::Providers::Vmware::InfraManager::SelectorSpec.selected_property?(type, p) }
-    change_set.reject!    { |c| !ManageIQ::Providers::Vmware::InfraManager::SelectorSpec.selected_property?(type, c["name"]) }
```

Comment:
We were using the change_set to do direct updates before (https://github.com/ManageIQ/manageiq/pull/14228/files#diff-a4716a448cf64d729b3428698d766ad4L36), without that it isn't used.